app/src/main/java/com/github/libretube/ui/views/CustomExoPlayerView.kt

```diff
@@ -202,6 +205,22 @@ internal class CustomExoPlayerView(
             override fun onScrubStop(timeBar: TimeBar, position: Long, canceled: Boolean) {}
         })
 
+        setControllerVisibilityListener(
+            ControllerVisibilityListener { visibility ->
+                playerViewModel?.isFullscreen?.value?.let { isFullscreen ->
+                    if (!isFullscreen) return@let
+                    when (visibility) {
+                        View.VISIBLE -> {
+                            activity.showSystemBars(WindowInsetsCompat.Type.statusBars())
+                        }
+                        View.GONE -> {
+                            activity.hideSystemBars(WindowInsetsCompat.Type.statusBars())
+                        }
+                    }
```

Comment:
```suggestion
                    val windowInsetsControllerCompat = WindowCompat.getInsetsController(activity.window, activity.window.decorView)
                    windowInsetsControllerCompat.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
                    if (visibility == View.VISIBLE) {
                        windowInsetsControllerCompat.show(WindowInsetsCompat.Type.statusBars())
                    } else {
                        windowInsetsControllerCompat.hide(WindowInsetsCompat.Type.statusBars())
                    }
```

Reply:
What's wrong with when

Reply:
According to the documentation for the listener, only View.VISIBLE and View.GONE are passed, so this is simpler.